pkg/auth/spire/certificate_provider.go

```diff
@@ -72,3 +75,17 @@ func (s *SpireDelegateClient) ValidateIdentity(identity string, cert *x509.Certi
 
 	return cert.URIs[0].String() == spiffeID, nil
 }
+
+func (s *SpireDelegateClient) NumericIdentityToSNI(id identity.NumericIdentity) string {
+	return fmt.Sprintf("%s.%s", id.String(), s.cfg.SpiffeTrustDomain)
```

Comment:
Can we do a simple string concat where we can instead of using `fmt.Sprintf`? It usually comes up in flamegraphs. I'm not sure how hot this code path is, but we might as well avoid it since it's trivial to and not have to deal with optimizing this later.